packages/legacy-workbench/src/js/angular/core/services/jwt-auth.service.js

```diff
@@ -66,12 +68,13 @@ angular.module('graphdb.framework.core.services.jwtauth', [
                 // remember where we were so we can return there
                 jwtAuth.updateReturnUrl();
 
-                $location.path('/login');
-                if (noaccess) {
-                    $location.search('noaccess');
-                } else if (expired) {
-                    $location.search('expired');
-                }
+                $location
+                    .path('/login')
+                    .search({
+                        ...($rootScope.returnToUrl ? {r: encodeURIComponent($rootScope.returnToUrl)} : {}),
```

Comment:
This is twisted.
Not sure if it is more readable than 3 separate ifs

Reply:
refactored